geodiff/src/geodiff.cpp

```diff
@@ -60,184 +60,93 @@ void GEODIFF_setMaximumLoggerLevel( GEODIFF_LoggerLevel maxLogLevel )
-      Logger::instance().error( "Modified does contain different number of tables than base in GEODIFF_createChangeset" );
-      return GEODIFF_UNSUPPORTED_CHANGE;
-    }
-
-    for ( size_t i = 0; i < mainTableNames.size(); ++i )
-    {
-      const std::string &table = mainTableNames.at( i );
-      const std::string &auxTable = auxTableNames.at( i );
-      if ( auxTable != table )
-      {
-        Logger::instance().error( "Modified renamed table " + table + " to " + auxTable + " in GEODIFF_createChangeset" );
-        return GEODIFF_UNSUPPORTED_CHANGE;
-      }
-
-      std::string errMsg;
-      bool hasSameSchema = has_same_table_schema( db, table, errMsg );
-      if ( !hasSameSchema )
-      {
-        Logger::instance().error( errMsg + " in GEODIFF_createChangeset" );
-        return GEODIFF_UNSUPPORTED_CHANGE;
-      }
-
-      int rc = sqlite3session_attach( session.get(), table.c_str() );
-      if ( rc )
-      {
-        Logger::instance().error( "Unable to attach session to database in GEODIFF_createChangeset" );
-        return GEODIFF_ERROR;
-      }
-      rc = sqlite3session_diff( session.get(), "aux", table.c_str(), NULL );
-      if ( rc )
-      {
-        Logger::instance().error( "Unable to diff tables in GEODIFF_createChangeset" );
-        return GEODIFF_ERROR;
-      }
-    }
-
-    Buffer changesetBuf;
-    changesetBuf.read( session );
-    changesetBuf.write( changeset );
+    std::map<std::string, std::string> conn;
+    conn["base"] = std::string( base );
+    conn["modified"] = std::string( modified );
+    if ( driverExtraInfo )
+      conn["conninfo"] = std::string( driverExtraInfo );
+    std::unique_ptr<Driver> driver( Driver::createDriver( std::string( driverName ) ) );
+    if ( !driver )
+      throw GeoDiffException( "Unable to use driver: " + std::string( driverName ) );
+    driver->open( conn );
 
-    return GEODIFF_SUCCESS;
+    ChangesetWriter writer;
+    if ( !writer.open( changeset ) )
+      throw GeoDiffException( "Unable to open changeset file for writing: " + std::string( changeset ) );
+    driver->createChangeset( writer );
   }
   catch ( GeoDiffException exc )
   {
     Logger::instance().error( exc );
     return GEODIFF_ERROR;
   }
-}
 
-static int conflict_callback( void *ctx, int conflict, sqlite3_changeset_iter *iterator )
-{
-  int *nconflicts = static_cast<int *>( ctx );
-  *nconflicts += 1;
-  std::string s = conflict2Str( conflict );
-  std::string vals = GeoDiffExporter::toString( iterator );
-  Logger::instance().warn( "CONFLICT: " + s  + ": " + vals );
-  return SQLITE_CHANGESET_REPLACE;
+  return GEODIFF_SUCCESS;
 }
 
-int GEODIFF_applyChangeset( const char *base, const char *changeset )
-{
-  if ( !base || !changeset )
-  {
-    Logger::instance().error( "NULL arguments to GEODIFF_applyChangeset" );
-    return GEODIFF_ERROR;
-  }
 
-  if ( !fileexists( base ) || !fileexists( changeset ) )
+int GEODIFF_applyChangesetEx( const char *driverName, const char *driverExtraInfo,
```

Comment:
do we want to keep the backward compatibity? in that case we can use the original functions with an extra driver parameter...